src/main/java/com/ldtteam/structurize/client/gui/AbstractBlueprintManipulationWindow.java

```diff
@@ -412,22 +418,93 @@ public boolean isAllowedCharacter(final char c)
 
                     if (testResult)
                     {
-                        // need properly typed thing now, but it's validated so parsing should never crash
-                        // TODO: switch instanceof
-                        final Object oldValue = typedSetting.get();
-                        if (oldValue instanceof Integer)
-                        {
-                            Structurize.getConfig().set(typedSetting, newValue.intValue());
-                        }
-                        else if (oldValue instanceof Long)
+                        inputField.setTextColor(0xffe0e0e0); // vanilla defualt
+
+                        final DoubleValue rendererTransparency = Structurize.getConfig().getClient().rendererTransparency;
+                        if (setting == rendererTransparency && rendererTransparency.get() < 0)
                         {
-                            Structurize.getConfig().set(typedSetting, newValue.longValue());
+                            // TODO: move to standalone ui
```

Comment:
do you want to address those todos before merging? =D

Reply:
need blockui pr first :D plus its small window, can wait for porting, any other todos expect this?